### PR TITLE
Feature/Category model

### PR DESCRIPTION
**Pull request by OlegMR0**

### CODE REVIEW

File: src/main/java/com/example/bookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public SecurityFilterChain web(HttpSecurity http) throws Exception {
                 .cors(AbstractHttpConfigurer::disable)
                 .csrf(AbstractHttpConfigurer::disable)
                 .authorizeHttpRequests((authorize) -> authorize
-                        .requestMatchers("/auth/**", "/error",
+                        .requestMatchers("/auth/**", "/error","/auth/login",
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/config/SecurityConfig.java]: /auth/login is redundant

File: src/main/java/com/example/bookstore/controller/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Operation(summary = "Delete a category by id",
+            description = "Delete an existing category with the provided id.")
+    @DeleteMapping("/{id}")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    public void deleteCategory(@PathVariable Long id) {
+        categoryService.deleteById(id);
+    }
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/controller/CategoryController.java]:
```suggestion

```

File: src/main/java/com/example/bookstore/dto/mapper/UserMapper.java [hunk 3]
```diff
@@ -5,9 +5,13 @@
...
 import org.mapstruct.Mapper;
 import org.mapstruct.NullValueCheckStrategy;
+import org.mapstruct.ReportingPolicy;
 
-@Mapper(componentModel = "spring", nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS)
+@Mapper(componentModel = "spring",
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        unmappedTargetPolicy = ReportingPolicy.IGNORE)
 public interface UserMapper {
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/dto/mapper/UserMapper.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/model/Category.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+@Table(name = "categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(unique = true, nullable = false)
+    private String name;
+    private String description;
+    private boolean isDeleted = false;
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/model/Category.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/service/CategoryService.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    List<CategoryResponseDto> findAll(Pageable pageable);
+
+    CategoryResponseDto getById(Long id);
+
+    CategoryResponseDto save(CreateCategoryRequestDto requestDto);
+
+    CategoryResponseDto update(Long id, CreateCategoryRequestDto requestDto);
+
+    void deleteById(Long id);
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/service/CategoryService.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/service/CategoryServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            Category updated = categoryRepository.save(category);
+            return categoryMapper.toDto(updated);
+        }
+        throw new EntityNotFoundException();
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        if (!categoryRepository.existsById(id)) {
+            throw new EntityNotFoundException();
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/service/CategoryServiceImpl.java]: add message + id

File: src/main/java/com/example/bookstore/service/CategoryServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Override
+    public CategoryResponseDto update(Long id, CreateCategoryRequestDto requestDto) {
+        if (categoryRepository.existsById(id)) {
+            Category category = categoryMapper.toCategory(requestDto);
+            category.setId(id);
+            Category updated = categoryRepository.save(category);
+            return categoryMapper.toDto(updated);
+        }
+        throw new EntityNotFoundException();
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/service/CategoryServiceImpl.java]: throw exception early (with message + id), unwrap rest of the code

File: src/main/java/com/example/bookstore/service/CategoryServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+@AllArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private CategoryRepository categoryRepository;
+    private CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryResponseDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable)
+                .stream()
+                .map(c -> categoryMapper.toDto(c))
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/service/CategoryServiceImpl.java]: avoid one letter naming in stream chains, change to category

File: src/main/java/com/example/bookstore/security/JwtService.java [hunk 9]
```diff
@@ -1,68 +1,68 @@
...
+import java.util.function.Function;
+import javax.crypto.SecretKey;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class JwtService {
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/security/JwtService.java]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]

**Comment by OlegMR0** [comment]
image
Do you think it's a good decision I've made handling lazy initialization? Or is there a better way?

**Review by okuzan — APPROVED** [review]
через join fetch це ок варіант, можеш через @EntityGraph(attributePaths = {"categories"})

**Pull request merged**